build-time-compiler/src/main/javadoc/README.md

```diff
@@ -0,0 +1,4 @@
+# Notice: Private API
```

Comment:
`build-time-compiler` is one of the gray areas ... we want to expose a stable enough API for integrators (like the Gradle plugin), I don't want to scare ppl away with this message.